### DISCUSSION TITLE
`Selectable` derive with aliases?

### DISCUSSION THREAD

**Discussion by ColonelThirtyTwo** [diesel-rs/diesel#3578] [Q&A]
Is there a way to use a `Selectable` type to select from an alias for the table specified by the Selectable implementation?

Example program:

```rust
mod schema;

use diesel::prelude::*;

use schema::foo;

diesel::alias!{
	schema::foo as bar: Bar,
}

#[derive(Debug, PartialEq, Clone, Queryable, Selectable)]
#[diesel(table_name=schema::foo)]
struct Data {
    id: i32,
    data: String,
}

fn main() {
    let mut db = diesel::SqliteConnection::establish("./test.sqlite").unwrap();

    foo::table
        .filter(schema::foo::id.eq(123))
        .select(Data::as_select())
        .load(&mut db)
        .unwrap();

    /*bar
        .filter(bar.field(schema::foo::id).eq(123))
        .select(Data::as_select())
        .load(&mut db)
        .unwrap();*/
}
```

With schema:

```sql
CREATE TABLE foo(
	id INTEGER PRIMARY KEY NOT NULL,
	data TEXT NOT NULL
);
```

The commented-out section of the rust code fails to compile, as `Data`'s `Expression` is tied to the `foo` table and not the `bar` alas it needs to use. `foo.fields` does not work, as the expression isn't a list of fields.

I could list the fields manually and use `alias.fields` to translate them, but then I'm duplicating what the `Selectable` derive is supposed to be doing for me. I could copy the `Data` struct with a `Selectable` implementation that uses the alias, but then I have two different, incompatible structures for the same data.

Is there something I'm missing that could translate the `Selectable`'s `Expression` to use the alias, or is this not implemented yet?

**Comment by weiznich** [comment]
Something like `.select(alias.fields(Data::construct_selection()))` might work there. Otherwise you need to provide your own wrapper there, although I'm not entirely sure if that's possible outside of diesel and how that exactly would look like.
